pydra/engine/core.py

```diff
@@ -563,25 +574,14 @@ def add(self, task):
             else:
                 task.state = state.State(task.name, other_states=other_states)
         self.node_names.append(task.name)
+        logger.debug("Added %s", task)
         return self
 
     def _run_task(self):
-        for task in self.graph_sorted:
-            # depend on prior tasks that have state
-            task.inputs.retrieve_values(self)
-            if task.state and not hasattr(task.state, "states_ind"):
-                task.state.prepare_states(inputs=task.inputs)
-            if task.state and not hasattr(task.state, "inputs_ind"):
-                task.state.prepare_inputs()
-            if self.plugin is None:
-                task.run()
-            else:
-                from .submitter import Submitter
-
-                with Submitter(self.plugin) as sub:
-                    sub.run(task)
-                while not task.done:
-                    sleep(1)
+        # logic in submitter.run
+        # TODO: enable lockfile for workflow execution
```

Comment:
this should automatically happen, since `workflow.run()` or `workflow()` will be called somewhere. shouldn't have to worry about that.